ui-frontend/packages/catalog-ui-search/src/main/webapp/component/query-basic/query-basic.view.tsx

```diff
@@ -260,59 +281,186 @@ function getFilterTree(model: any) {
-    return (
-      <>
-        <div className="editor-properties">
-          <div
-            className="basic-text"
-            data-help="Search by free text using the
-    grammar of the underlying source.
-    For wildcard searches, use * after or before partial keywords (e.g. *earth*)."
+type CurrentValueType = { text: string; matchcase: boolean; basicTime: any }
+
+type QueryBasicProps = {
+  model: any
+}
+
+const constructFilterFromBasicFilter = ({
+  basicFilter,
+}: {
+  basicFilter: {
+    anyText: Array<FilterClass>
+    anyDate: Array<BasicFilterClass>
+    anyGeo: Array<FilterClass>
+  }
+}) => {
+  const filters = []
+  if (basicFilter.anyText[0].value !== '') {
+    filters.push(basicFilter.anyText[0])
+  }
+
+  if (basicFilter.anyDate[0] !== undefined) {
+    filters.push({
+      type: 'OR',
+      filters: basicFilter.anyDate[0].property.map(property => {
+        return {
+          ...basicFilter.anyDate[0],
+          property,
+        }
+      }),
+    })
+  }
+
+  if (filters.length === 0) {
+    filters.unshift(CQLUtils.generateFilter('ILIKE', 'anyText', '*'))
+  }
+
+  return {
+    type: 'AND',
+    filters,
+  }
+}
+
+const QueryBasic = ({ model }: QueryBasicProps) => {
+  const [basicFilter, setBasicFilter] = React.useState(
+    translateFilterToBasicMap(getFilterTree(model)).propertyValueMap
+  )
+
+  const { listenTo, stopListening } = useBackbone()
+  const saveCallbackRef = React.useRef(() => {
+    model.set('cql', cql.write(constructFilterFromBasicFilter({ basicFilter })))
+    model.set('filterTree', constructFilterFromBasicFilter({ basicFilter }))
+  })
+  React.useEffect(
+    () => {
+      saveCallbackRef.current = () => {
+        model.set(
+          'cql',
+          cql.write(constructFilterFromBasicFilter({ basicFilter }))
+        )
+        model.set('filterTree', constructFilterFromBasicFilter({ basicFilter }))
+      }
+    },
+    [basicFilter, model]
+  )
+  React.useEffect(() => {
+    return () => {
+      saveCallbackRef.current()
+    }
+  }, [])
+  React.useEffect(
+    () => {
+      const callback = () => {
+        saveCallbackRef.current()
+      }
+      const callback2 = () => {
+        setBasicFilter(
+          translateFilterToBasicMap(getFilterTree(model)).propertyValueMap
+        )
+      }
+      // for perf, only update when necessary
+      listenTo(model, 'update', callback)
+      listenTo(model, 'change:filterTree', callback2)
+      return () => {
+        stopListening(model, 'update', callback)
+        stopListening(model, 'change:filterTree', callback2)
+      }
+    },
+    [model, basicFilter]
+  )
+  return (
+    <>
+      <div className="editor-properties px-2 py-3">
+        <div className="">
+          <Typography className="pb-2">Text</Typography>
+          <TextField
+            fullWidth
+            value={basicFilter.anyText ? basicFilter.anyText[0].value : ''}
+            placeholder={`Text to search for. Use "*" for wildcard.`}
```

Comment:
❓ I'm may just be missing what this will look like, but is "Text to search for." intentional?

Reply:
I copied the placeholder we had from the previous, open to changing it.